app/models/facility_district.rb

```diff
@@ -0,0 +1,37 @@
+class FacilityDistrict
+  attr_reader :name, :scope
+
+  alias_method :id, :name
+  alias_method :to_param, :name
+
+  def initialize(name:, scope: Facility.all)
+    @name = name
+    @scope = scope
+  end
+
+  def facilities
+    scope.where(district: name)
+  end
+
+  def dashboard_analytics(period:, prev_periods:)
+    query = DistrictAnalyticsQuery.new(name, facilities, period, prev_periods, include_current_period: true)
+    query.call
+  end
+
+  def cohort_analytics(period:, prev_periods:)
```

Comment:
So these query calls will piggyback on the cache mechanism in the queries?
Nice!

Reply:
Yup, just like they do on `FacilityGroup` and `Facility` already! 😄